tests/compiletest.rs

```diff
@@ -1,82 +1,123 @@
+use colored::*;
+use regex::Regex;
 use std::env;
 use std::path::PathBuf;
-
-use colored::*;
-use compiletest_rs as compiletest;
+use ui_test::{Config, Mode};
 
 fn miri_path() -> PathBuf {
     PathBuf::from(option_env!("MIRI").unwrap_or(env!("CARGO_BIN_EXE_miri")))
 }
 
-fn run_tests(mode: &str, path: &str, target: &str) {
+fn run_tests(mode: Mode, path: &str, target: Option<String>) {
     let in_rustc_test_suite = option_env!("RUSTC_STAGE").is_some();
+
+    if in_rustc_test_suite {
+        std::env::set_var("MIRI_SKIP_UI_CHECKS", "1");
```

Comment:
Given that `env::set_var` is a troubled API, and that the intended recipient of this flag is this very same function -- can you avoid this, and instead `set` this in the rustc bootstrap code?